src/service/service.py

```diff
@@ -66,14 +68,33 @@ async def lifespan(app: FastAPI) -> AsyncGenerator[None, None]:
 
 
 def _parse_input(user_input: UserInput) -> tuple[dict[str, Any], UUID]:
-    run_id = uuid4()
+    run_id = user_input.run_id or uuid4()
     thread_id = user_input.thread_id or str(uuid4())
+    if user_input.type == "user":
+        inputU = {"messages": [HumanMessage(content=user_input.message)]}
+    else:
+        if user_input.message == "aproved_tool":
+            inputU = None # This means to continue graph execution after interruption
+        elif user_input.message == "canceled_tool":
+            inputU = {"messages": [ToolMessage(
+                        tool_call_id=user_input.tool_call_id,
+                        content=f"API call denied by user. Ask them about de reason or what else can you help them with.",
+                        )]}
+        else:
+            inputU = {"messages": [
+                            ToolMessage(
+                                tool_call_id=user_input.tool_call_id,
+                                content=f"API call denied by user.",
+                                ),
+                            HumanMessage(content=user_input.message)
+                        ]}
+
     kwargs = {
-        "input": {"messages": [HumanMessage(content=user_input.message)]},
-        "config": RunnableConfig(
-            configurable={"thread_id": thread_id, "model": user_input.model}, run_id=run_id
-        ),
-    }
+            "input": inputU,
+            "config": RunnableConfig(
+                configurable={"thread_id": thread_id, "model": user_input.model}, run_id=run_id
+            ),
+        }
```

Comment:
You'll want to set up the pre-commit hooks and run the linter and tests locally. Otherwise the CI will keep failing and won't be able to merge this. See the contributing instructions in the README.